lgc/patch/RegisterMetadataBuilder.cpp

```diff
@@ -172,6 +173,36 @@ void RegisterMetadataBuilder::buildPalMetadata() {
     if (hwStageMask & (Util::Abi::HwShaderGs | Util::Abi::HwShaderVs))
       buildPaSpecificRegisters();
 
+    if (lastVertexProcessingStage != ShaderStageInvalid) {
+      // Fill ".preraster_output_semantic"
+      auto resUsage = m_pipelineState->getShaderResourceUsage(lastVertexProcessingStage);
+      auto &outputLocInfoMap = resUsage->inOutUsage.outputLocInfoMap;
+      msgpack::ArrayDocNode *preRasterOutputSemanticNode =
+          outputLocInfoMap.empty()
+              ? nullptr
+              : &getPipelineNode()[Util::Abi::PipelineMetadataKey::PrerasterOutputSemantic].getArray(true);
```

Comment:
In this case, I think it's worth to just add another `if` to guard all of this by `!outputLocInfoMap.empty() || !builtInOutputLocMap.empty()`. Then the slightly strange ternary operator here can be avoided.

Reply:
Fixed.